src/app/_components/post-editor.tsx

```diff
@@ -0,0 +1,83 @@
+"use client";
+
+import { api } from "~/trpc/react";
+import { Avatar, AvatarImage } from "./shadcn-ui/avatar";
+import { Button } from "./shadcn-ui/button";
+import { Separator } from "./shadcn-ui/separator";
+import { useState, type ChangeEvent } from "react";
+import { useToast } from "./shadcn-ui/use-toast";
+import { useRouter } from "next/navigation";
+
+interface Properties {
+  avatar: string;
+  username: string;
+}
+
+export default function PostEditor(props: Properties) {
+  const router = useRouter();
+  const { toast } = useToast();
+  const [input, setInput] = useState("");
+
+  const onChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
+    const element = event.target;
+
+    // Automatically resizes textarea height to fit text
+    element.style.height = "0px";
+    element.style.height = element.scrollHeight + "px";
+
+    setInput(element.value);
+  };
+
+  const createPost = api.posts.create.useMutation({
```

Comment:
```suggestion
  const { mutation, isLoading } = api.posts.create.useMutation({
```

Use the `isLoading` property to add a loading display while a post is being processed.

Note that the `mutation` property is just the function itself.